client/editor/map_editor.cpp

```diff
@@ -111,3 +121,40 @@ void map_editor::close()
   queen()->mapview_wdg->show_all_fcwidgets();
   queen()->unitinfo_wdg->show();
 }
+
+/**
+ * \brief Populate a combo box with the current list of players
+ */
+void map_editor::update_players()
+{
+  if (!players_done) {
+    players_iterate(pplayer)
+    {
+      auto sprite =
+          *get_nation_flag_sprite(tileset, nation_of_player(pplayer));
+      ui.combo_players->addItem(sprite,
+                                _(qUtf8Printable(player_name(pplayer))));
+    }
+    players_iterate_end;
+    players_done = true;
+  }
+}
+
+/**
+ * \brief Activate the Tile Tool when clicked
+ */
+void map_editor::select_tool_tile()
+{
+  if (ett_wdg_active) {
+    ui.tbut_edit_tile->setToolTip(_("Open Tile Tool"));
+    ett_wdg->hide();
+    ett_wdg_active = false;
+  } else {
+    ui.tbut_edit_tile->setToolTip(_("Close Tile Tool"));
+    ett_wdg = new editor_tool_tile(0);
```

Comment:
```suggestion
    ett_wdg = new editor_tool_tile;
```
Background: by convention, the parent is optional and the default is `nullptr`.